OISF/suricata
# Pcre2 3194 v20

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3194
https://redmine.openinfosecfoundation.org/issues/4446

Describe changes:
- migrate all pcre1 code to pcre2
- qa : remove some obsolete files
- pcrexform : checks that we have exactly one substring capture in a signature

suricata-verify-pr: 502

https://github.com/OISF/suricata-verify/pull/502

Replaces https://github.com/OISF/suricata/pull/6394 with rebase and last commit to remove pcre1 not including GitHub CI yet
